### ISSUE TITLE
Refactor support

### ISSUE THREAD

**Issue by prabhu-madanagopal** [Ruin0x11/lsp-intellij#10]
Is there a plan to support IntelliJ refactor 'Extract method, variable, field, constant' and other refactor in lsp-intellij and server?

**Comment by Ruin0x11** [comment]
Yes, I was thinking maybe the server could support such actions by using the `Code Action` feature of LSP at some point. I haven't done work on it yet, though.
